### DISCUSSION TITLE
Reusable popover/tooltip components

### DISCUSSION THREAD

**Discussion by switz** [floating-ui/floating-ui#2429] [React]
I see that there are code sandboxes for reusable popover/tooltip components. Since this is a low level library, are there any maintained higher level libraries on npm for tooltips/popups?

Would be nice for those of us who just need simple interactions and don't want to muck around with lower level APIs.

```
<Popover content={<div>hi</div>}>click me</Popover>
<Tooltip content={<div>hi</div>}>hover me</Tooltip>
```

https://floating-ui.com/docs/popover#reusable-popover-component

**Comment by atomiks** [comment]
The components are written to be mostly copy paste-able with some minor edits needed to customize them. You can copy the entire content of the file into your app and use it freely.

The upside of this is that you get high control/flexibility when you want behavior to change a bit. The minor downside is not receiving updates for the component itself - but since the lower-level pieces are still abstracted away and get updates, this trade-off is usually worth it and quite maintainable because the components themselves are pretty small.
